Add mLUKE

ryokan0123 opened the pull request:
# What does this PR do?
This PR adds mLUKE, [mLUKE: The Power of Entity Representations in Multilingual Pretrained Language Models](https://arxiv.org/abs/2110.08151), by Studio Ousia.

The model is a multilingual extension of [LUKE](https://huggingface.co/transformers/model_doc/luke.html), which is trained on the Wikipedia articles in 24 languages.
We have uploaded the model weights on [the hugging face hub](https://huggingface.co/studio-ousia/mluke-base).

The only difference from LUKE is the tokenizer, so I’ve just added `MLukeTokenizer`, which is a mix of `LukeTokenizer` and `XLMRobertaTokenizer`.

I’ve also added `LukeForMaskedLM` to perform the cloze prompt task experimented in [the original paper](https://arxiv.org/abs/2110.08151).
The example can be found in [this notebook](https://colab.research.google.com/drive/1FAJimyuzUefZyWS1ckSEitglLfFRofi4?hl=ja#scrollTo=jIXV3onW2ITm).

## Who can review?
@NielsRogge, could you help us review the code as you have worked on LUKE?🙂

ryokan0123 committed:
add MLukeTokenizer

ryokan0123 committed:
add MLukeTokenizer

ryokan0123 committed:
add MLukeTokenizer

ryokan0123 committed:
add MLukeTokenizer

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
debug

ryokan0123 committed:
debug

ryokan0123 committed:
debug

ryokan0123 committed:
debug

ryokan0123 committed:
debug

ryokan0123 committed:
debug

ryokan0123 committed:
update

ryokan0123 committed:
update

ryokan0123 committed:
update

ryokan0123 committed:
add assertions in convert_mluke_original_pytorch_checkpoint_to_pytorch.py

ryokan0123 committed:
add assertions in convert_mluke_original_pytorch_checkpoint_to_pytorch.py

ryokan0123 committed:
add assertions in convert_mluke_original_pytorch_checkpoint_to_pytorch.py

ryokan0123 committed:
update tokenizatio_auto

ryokan0123 committed:
update tokenizatio_auto

ryokan0123 committed:
update tokenizatio_auto

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
pass the tests

ryokan0123 committed:
pass the tests

ryokan0123 committed:
pass the tests

ryokan0123 committed:
add documentation

ryokan0123 committed:
add documentation

ryokan0123 committed:
add documentation

ryokan0123 committed:
add docstrings

ryokan0123 committed:
add docstrings

ryokan0123 committed:
add docstrings

ryokan0123 committed:
format code

ryokan0123 committed:
format code

ryokan0123 committed:
format code

ryokan0123 committed:
fix docstrings

ryokan0123 committed:
fix docstrings

ryokan0123 committed:
fix docstrings

ryokan0123 committed:
add LukeForMaskedLM

ryokan0123 committed:
add LukeForMaskedLM

ryokan0123 committed:
add LukeForMaskedLM

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
update the conversion script

ryokan0123 committed:
update the conversion script and model

ryokan0123 committed:
update the conversion script and model

ryokan0123 committed:
update the conversion script and model

ryokan0123 committed:
fixup

ryokan0123 committed:
fixup

ryokan0123 committed:
fixup

ryokan0123 committed:
minor fix in comments

ryokan0123 committed:
minor fix in comments

ryokan0123 committed:
minor fix in comments

ryokan0123 committed:
fix documentation

ryokan0123 committed:
fix documentation

ryokan0123 committed:
fix documentation

ryokan0123 committed:
fix format

ryokan0123 committed:
fix format

ryokan0123 committed:
fix format

ryokan0123 committed:
get back pyproject.toml

ryokan0123 committed:
get back pyproject.toml

ryokan0123 committed:
get back pyproject.toml

ryokan0123 committed:
get back pyproject.toml

gsnidero committed:
Add documentation for multi-label classification (#14168)

* "update example docstring multilabel example

* update example docstring multilabel example

ryokan0123 committed:
fix test by removing use_auth_token

ryokan0123 committed:
fix test by removing use_auth_token

ryokan0123 committed:
fix test by removing use_auth_token

ryokan0123 committed:
update doc

ryokan0123 committed:
update doc

ryokan0123 committed:
update doc

t-vi committed:
use functional interface for softmax in attention (#14198)

* use functional interface instead of instantiating module and immediately calling it

* fix torch.nn.functional to nn.functional. Thank you Stas!

patil-suraj committed:
VisionTextDualEncoder (#13511)

* init vision_text_dual_encoder

* fix merge

* remove extra heads

* fix tests

* remove VISION_TEXT_DUAL_ENCODER_PRETRAINED_CONFIG_ARCHIVE_MAP

* remove archive map

* fix imports

* fix more imports

* fix init

* delete tokenizers

* fix imports

* clean

* support clip's vision model

* handle None config

* begin tests

* more test and few fixes

* warn about newly init weights

* more tests

* add loss to model

* remove extra classes from doc

* add processor

* doc and small fixes

* add start docstr

* update flax model

* flax tests

* more flax tests

* doc

* quality

* doc and quality

* fix doc

* doc

* remove comments

* update warning

* quality

* fix docs

* Apply suggestions from code review

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

* replace asserts, fix imports

* update imports

* fix import

* address some review comments

* fix check

* reduce tolerance

* fix test

* add flax integration test

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* address Sylvain's comments

* fix style

* add pt_flax_equivalence test in PT tests

* add pt integration test

* update test

* use pre-trained checkpoint in examples

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

JamesDeAntonis committed:
WIP: Support for Training with BF16 (#13207)

* started bf16 integration

* minor changes

* code now runs

* style

* lay foundation for bf16 testing

* lay foundation for bf16 testing

* start the tests

* better bf16 check

* style

* 2 separate checkers - one for bf16 support, another for bf16+autocast

* Update src/transformers/training_args.py

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

* a couple of comment resolutions

* more comment resolutions

* resolved a small bug

* just some print statemtns

* added todo marking

* added a todo

* adjust for API change s/fast_dtype/dtype/

* fix style

* merge 2 bf16 util functions

* bf16 now does scaling too

* Add support for bfloat16

* Revert T5 layernorm to float32

This is based on the comment at https://github.com/huggingface/transformers/pull/14448/files#r752660929 and the PyTorch PR https://github.com/pytorch/pytorch/pull/66920 .

* Add comment about conversion to float32 before returning the numpy data

* Add comment about AMP-bfloat16 incompatibility

* Fix formatting

* typo

* reformer / bf16

* cleanup

* require at least pt-1.10

* fix

* will deal with deepspeed separately

* cleanup

* revert

* cleanup

* fp16_full_eval and bf16_full_eval are separate modes

* proper deprecation

* cleanup

* test and fixes

* spelling

* cleanup

* add a note that this API is experimental

Co-authored-by: jamie <jamie@cortx.com>
Co-authored-by: Stas Bekman <stas@stason.org>
Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>
Co-authored-by: suriya <suriya@cortx.com>
Co-authored-by: Manuel R. Ciosici <manuelrciosici@gmail.com>

ryokan0123 committed:
format

ryokan0123 committed:
format

ryokan0123 committed:
format

ryokan0123 committed:
format

ryokan0123 committed:
format

ryokan0123 committed:
format

ryokan0123 committed:
format

ryokan0123 committed:
format

patil-suraj committed:
FlaxGPTJ (#14396)

* add flax gptj

* no bias in attention dense

* no wpe

* fix rotary embeddings

* fix rotary embeds

* fix rotray embeds

* quality

* doc and quality

* fix equivalence tests

ryokan0123 committed:
fix typo

ryokan0123 committed:
fix typo

ryokan0123 committed:
fix typo

NielsRogge commented on src/transformers/models/mluke/__init__.py:
```diff
@@ -0,0 +1,35 @@
+from ...file_utils import _LazyModule
+
+
+_import_structure = {
+    "tokenization_mluke": ["MLukeTokenizer"],
+}
```
```suggestion
if is_sentencepiece_available():
    _import_structure["tokenization_mluke"]: ["MLukeTokenizer"]
```

Ryou0634 replied:
Yes, it does. `is_sentencepiece_available()` should be added. Thanks!

NielsRogge commented on src/transformers/models/mluke/__init__.py:
```diff
@@ -0,0 +1,35 @@
+}
+
+if TYPE_CHECKING:
+    from .tokenization_mluke import MLukeTokenizer
```
```suggestion
    if is_sentencepiece_available():
        from .tokenization_mluke import MLukeTokenizer
```

NielsRogge commented on src/transformers/models/mluke/convert_mluke_original_pytorch_checkpoint_to_pytorch.py:
```diff
@@ -0,0 +1,228 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Convert LUKE checkpoint."""
```
```suggestion
"""Convert mLUKE checkpoint."""
```

NielsRogge commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1763 @@
+# coding=utf-8
+# Copyright 2018 Google AI, Google Brain and Carnegie Mellon University Authors and the HuggingFace Inc. team.
```
```suggestion
# Copyright 2021 Studio Ousia and the HuggingFace Inc. team.
```

NielsRogge commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1763 @@
+            )
+
+    @add_end_docstrings(ENCODE_KWARGS_DOCSTRING, ENCODE_PLUS_ADDITIONAL_KWARGS_DOCSTRING)
+    def encode_plus(
```
Is it possible that the encode_plus, batch_encode_plus and encode methods can be removed, because can be passed as kwargs to the `_encode_plus` and `_batch_encode_plus` methods.

You can verify that everything still works when removing these methods.

cc @sgugger

Ryou0634 replied:
Yeah, taking a closer look, `encode_plus` and `batch_encode_plus` are almost the same as the super class's one.
After removing them, I confirmed that they still work.

I've removed the functions just from `LukeTokenizer` and `MLukeTokenizer`, and left the others as it is to make the PR consistent.

NielsRogge commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1763 @@
+        )
+
+    @add_end_docstrings(ENCODE_KWARGS_DOCSTRING, ENCODE_PLUS_ADDITIONAL_KWARGS_DOCSTRING)
+    def batch_encode_plus(
```
This method can potentially be removed, see comment above.

NielsRogge reviewed (commented):
Wow, this PR looks in great shape already! Amazing work.

There are only some minor things to be improved, but this can be merged soon.

stas00 committed:
fix autocast for older pytorch

sgugger committed:
Doc new front (#14590)

* Convert PretrainedConfig doc to Markdown

* Use syntax

* Add necessary doc files (#14496)

* Doc fixes (#14499)

* Fixes for the new front

* Convert DETR file for table

* Title is needed

* Simplify a bit

* Even simpler

* Remove imports

* Fix typo in toctree (#14516)

* Fix checkpoints badge

* Update versions.yml format (#14517)

* Doc new front github actions (#14512)

* Doc new front github actions

* Fix docstring

* Fix feature extraction utils import (#14515)

* Address Julien's comments

* Push to doc-builder

* Ready for merge

* Remove old build and deploy

* Doc misc fixes (#14583)

* Rm versions.yml from doc

* Fix converting.rst

* Rm pretrained_models from toctree

* Fix index links (#14567)

* Fix links in README

* Localized READMEs

* Fix copy script

* Fix find doc script

* Update README_ko.md

Co-authored-by: Julien Chaumond <julien@huggingface.co>

Co-authored-by: Julien Chaumond <julien@huggingface.co>

* Adapt build command to new CLI tools (#14578)

* Fix typo

* Fix doc interlinks (#14589)

* Convert PretrainedConfig doc to Markdown

* Use syntax

* Rm pattern <[a-z]+(.html).*>

* Rm huggingface.co/transformers/master

* Rm .html

* Rm .html from index.mdx

* Rm .html from model_summary.rst

* Update index.mdx rm html

* Update remove .html

* Fix inner doc links

* Fix interlink in preprocssing.rst

* Update pr_checks

Co-authored-by: Sylvain Gugger <sylvain.gugger@gmail.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Convert PretrainedConfig doc to Markdown

* Use syntax

* Add necessary doc files (#14496)

* Doc fixes (#14499)

* Fixes for the new front

* Convert DETR file for table

* Title is needed

* Simplify a bit

* Even simpler

* Remove imports

* Fix checkpoints badge

* Fix typo in toctree (#14516)

* Update versions.yml format (#14517)

* Doc new front github actions (#14512)

* Doc new front github actions

* Fix docstring

* Fix feature extraction utils import (#14515)

* Address Julien's comments

* Push to doc-builder

* Ready for merge

* Remove old build and deploy

* Doc misc fixes (#14583)

* Rm versions.yml from doc

* Fix converting.rst

* Rm pretrained_models from toctree

* Fix index links (#14567)

* Fix links in README

* Localized READMEs

* Fix copy script

* Fix find doc script

* Update README_ko.md

Co-authored-by: Julien Chaumond <julien@huggingface.co>

Co-authored-by: Julien Chaumond <julien@huggingface.co>

* Adapt build command to new CLI tools (#14578)

* Fix typo

* Fix doc interlinks (#14589)

* Convert PretrainedConfig doc to Markdown

* Use syntax

* Rm pattern <[a-z]+(.html).*>

* Rm huggingface.co/transformers/master

* Rm .html

* Rm .html from index.mdx

* Rm .html from model_summary.rst

* Update index.mdx rm html

* Update remove .html

* Fix inner doc links

* Fix interlink in preprocssing.rst

* Update pr_checks

Co-authored-by: Sylvain Gugger <sylvain.gugger@gmail.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Styling

Co-authored-by: Mishig Davaadorj <mishig.davaadorj@coloradocollege.edu>
Co-authored-by: Lysandre Debut <lysandre@huggingface.co>
Co-authored-by: Julien Chaumond <julien@huggingface.co>

qqaatw committed:
Fix mask token handling (#14364)

* Fix mask token handling

* Revert "Fix mask token handling"

This reverts commit daaa3f5291b1f71e5bc3604ca281c000000c4648.

* Fix FNet mask token tokenization

stas00 committed:
[doc] bf16/tf32 guide (#14579)

* [doc] bf16/tf32 guide

* expand

* expand

* Update docs/source/performance.md

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

ryokan0123 committed:
Apply suggestions from code review

add is_sentencepiece_available in __init__.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

ryokan0123 committed:
Apply suggestions from code review

add is_sentencepiece_available in __init__.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

ryokan0123 committed:
Apply suggestions from code review

add is_sentencepiece_available in __init__.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

ryokan0123 committed:
Apply suggestions from code review

fix typo

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

ryokan0123 committed:
Apply suggestions from code review

fix typo

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

ryokan0123 committed:
Apply suggestions from code review

fix typo

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

ryokan0123 committed:
add missing import

ryokan0123 committed:
add missing import

ryokan0123 committed:
add missing import

ryokan0123 committed:
remove encode_plus and batch_encode_plus from (M)LukeTokenizer

ryokan0123 committed:
remove encode_plus and batch_encode_plus from (M)LukeTokenizer

ryokan0123 committed:
remove encode_plus and batch_encode_plus from (M)LukeTokenizer

ryokan0123 committed:
fix import MLukeTokenizer

ryokan0123 committed:
fix import MLukeTokenizer

ryokan0123 committed:
fix import MLukeTokenizer

ryokan0123 committed:
fix import

ryokan0123 committed:
fix import

ryokan0123 committed:
fix import

ryokan0123 committed:
fix up

ryokan0123 committed:
fix up

ryokan0123 committed:
fix up

Commit:
Rename toctree.yml -> _toctree.yml (#14594)

Commit:
Update doc img links (#14593)

* Update doc img links

* Rename toctree.yml -> _toctree.yml (#14594)

* Update doc img links

* Update performance.md img link

ryokan0123 committed:
add mLUKE to multilingual.rst

ryokan0123 committed:
add mLUKE to multilingual.rst

ryokan0123 committed:
add mLUKE to multilingual.rst

Ryou0634 commented:
Thank you for reviewing, @NielsRogge!
I’ve reflected the reviews to the code and also added mLUKE to the Multilingual-models section in the doc.

ryokan0123 committed:
fixup multilingual.rst

ryokan0123 committed:
fixup multilingual.rst

ryokan0123 committed:
fixup multilingual.rst

LysandreJik committed:
Adds a git pull instruction to the documentation builder (#14597)

* Adds a git pull instruction

* master -> main

stancld committed:
[Flax] Add FlaxBlenderbotSmall (#14576)

* [WIP] Add FlaxBlenderbotSmall

* Revert some unintentionally changed files

Revert some unintentionally files changed by improperly filled cookiecutter instructions.

* Fix repo consistency

* Fix Flax-PT equivalence

* Apply suggestions from code review

* Update index.mdx

* Apply suggestions from code review

Co-authored-by: Suraj Patil <surajp815@gmail.com>

LysandreJik committed:
Python 3.6 -> Python 3.7 for TF runs (#14598)

lvwerra committed:
Add CodeParrot 🦜 codebase (#14536)

* add readme skeleton

* update readme

* add initialization script

* add deduplication script

* add codeparrot training script

* add code generation evaluation

* add validation loss script

* add requirements

* update readme

* tweak readme

* make style

* add highlights to readme

* add CLIs to scripts

* add tokenizer training script

* add docstring to constant length dataset

* fix defaults in arguments

* update readme with cli

* move image to hub

* tweaks of readme

* fix cli commands

* add author

* explain env variables

* fix formatting

* Update examples/research_projects/codeparrot/README.md

Co-authored-by: lewtun <lewis.c.tunstall@gmail.com>

* Apply suggestions from code review

Co-authored-by: lewtun <lewis.c.tunstall@gmail.com>

* replace generic with gpt2 tokenizer

Co-authored-by: lewtun <lewis.c.tunstall@gmail.com>

yis11178 committed:
change tf.math.divide with int(/) to remove dim_per_head from the TF graph (#14600)

Co-authored-by: yis <yis@graphcore.ai>

nikvaessen committed:
fix #14524 (IndexError when mask prob is too low) (#14525)

* fix #14524 (IndexError when mask prob is too low)

* fix formatting

* correct documentation, add option for setting min_num_masks

* change the semantic meaning of `mask_prob` in _compute_mask_indices

With this commit the meaing of `mask_prob` actually adhered to the probability for each
vector to be the start of a masked span of length.

* fix check_copies test

* fix documentation to semantic meaning of `upper bound of overall masking percentage`, revert changes to _compute_mask_indices

* fix typo

NielsRogge commented:
Thanks for taking into account the comments. Can you rebase with master? The index.rst file is now replaced by an [index.mdx](https://github.com/huggingface/transformers/blob/master/docs/source/index.mdx) file because of the new front-end website.

sgugger commented on docs/source/index.rst:
```diff
@@ -646,6 +646,7 @@ Flax), PyTorch, and/or TensorFlow.
     model_doc/led
     model_doc/longformer
     model_doc/luke
+    model_doc/mluke
```
Please also add your model to the README.

sgugger commented on docs/source/model_doc/mluke.rst:
```diff
@@ -0,0 +1,66 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The mLUKE model was proposed in `mLUKE: The Power of Entity Representations in Multilingual Pretrained Language Models
+<https://arxiv.org/abs/2110.08151>`_ by Ryokan Ri, Ikuya Yamada, and Yoshimasa Tsuruoka. It's a multilingual extension
```
```suggestion
<https://arxiv.org/abs/2110.08151>`__ by Ryokan Ri, Ikuya Yamada, and Yoshimasa Tsuruoka. It's a multilingual extension
```

sgugger commented on docs/source/model_doc/mluke.rst:
```diff
@@ -0,0 +1,66 @@
+
+The mLUKE model was proposed in `mLUKE: The Power of Entity Representations in Multilingual Pretrained Language Models
+<https://arxiv.org/abs/2110.08151>`_ by Ryokan Ri, Ikuya Yamada, and Yoshimasa Tsuruoka. It's a multilingual extension
+of the `LUKE model <https://arxiv.org/abs/2010.01057>`__ trained on the basis of XLMRoBERTa.
```
```suggestion
of the `LUKE model <https://arxiv.org/abs/2010.01057>`__ trained on the basis of XLM-RoBERTa.
```

sgugger commented on docs/source/model_doc/mluke.rst:
```diff
@@ -0,0 +1,66 @@
+<https://arxiv.org/abs/2110.08151>`_ by Ryokan Ri, Ikuya Yamada, and Yoshimasa Tsuruoka. It's a multilingual extension
+of the `LUKE model <https://arxiv.org/abs/2010.01057>`__ trained on the basis of XLMRoBERTa.
+
+It is based on XLMRoBERTa and adds entity embeddings, which helps improve performance on various downstream tasks
```
```suggestion
It is based on XLM-RoBERTa and adds entity embeddings, which helps improve performance on various downstream tasks
```

sgugger commented on src/transformers/models/luke/modeling_luke.py:
```diff
@@ -1013,6 +1088,176 @@ def create_position_ids_from_input_ids(input_ids, padding_idx):
+                loss = loss + mep_loss
+
+        if not return_dict:
+            output = (
+                logits,
+                entity_logits,
+                outputs.hidden_states,
+                outputs.entity_hidden_states,
+                outputs.attentions,
+            )
```
```suggestion
            output = (logits, entity_logits, outputs.hidden_states, outputs.entity_hidden_states, outputs.attentions)
```
This all fits in one line.

sgugger commented on src/transformers/models/mluke/convert_mluke_original_pytorch_checkpoint_to_pytorch.py:
```diff
@@ -0,0 +1,228 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Inc. team.
```
```suggestion
# Copyright 2021 The HuggingFace Inc. team.
```

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+        # Input type checking for clearer error
+        is_valid_single_text = isinstance(text, str)
+        is_valid_batch_text = isinstance(text, (list, tuple)) and (len(text) == 0 or (isinstance(text[0], str)))
+        assert (
+            is_valid_single_text or is_valid_batch_text
+        ), "text input must be of type `str` (single example) or `List[str]` (batch)."
```
No new assert in the codebase please (conversion script is fine but that's all). This should be an if statement and raise a `ValueError` when needed.

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+        is_valid_batch_text_pair = isinstance(text_pair, (list, tuple)) and (
+            len(text_pair) == 0 or isinstance(text_pair[0], str)
+        )
+        assert (
+            text_pair is None or is_valid_single_text_pair or is_valid_batch_text_pair
+        ), "text_pair input must be of type `str` (single example) or `List[str]` (batch)."
```
Same here

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+            if entity_spans is None:
+                first_ids = get_input_ids(text)
+            else:
+                assert isinstance(entity_spans, list) and (
+                    len(entity_spans) == 0 or isinstance(entity_spans[0], tuple)
+                ), "entity_spans should be given as a list of tuples containing the start and end character indices"
+                assert entities is None or (
+                    isinstance(entities, list) and (len(entities) == 0 or isinstance(entities[0], str))
+                ), "If you specify entities, they should be given as a list of entity names"
+                assert entities is None or len(entities) == len(
+                    entity_spans
+                ), "If you specify entities, entities and entity_spans must be the same length"
```
Same here

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+                if entity_spans_pair is None:
+                    second_ids = get_input_ids(text_pair)
+                else:
+                    assert isinstance(entity_spans_pair, list) and (
+                        len(entity_spans_pair) == 0 or isinstance(entity_spans_pair[0], tuple)
+                    ), "entity_spans_pair should be given as a list of tuples containing the start and end character indices"
+                    assert entities_pair is None or (
+                        isinstance(entities_pair, list)
+                        and (len(entities_pair) == 0 or isinstance(entities_pair[0], str))
+                    ), "If you specify entities_pair, they should be given as a list of entity names"
+                    assert entities_pair is None or len(entities_pair) == len(
+                        entity_spans_pair
+                    ), "If you specify entities_pair, entities_pair and entity_spans_pair must be the same length"
```
Same here.

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+                        second_entity_ids = [self.entity_vocab.get(entity, unk_entity_id) for entity in entities_pair]
+
+        elif self.task == "entity_classification":
+            assert (
+                isinstance(entity_spans, list) and len(entity_spans) == 1 and isinstance(entity_spans[0], tuple)
+            ), "Entity spans should be a list containing a single tuple containing the start and end character indices of an entity"
```
Same here.

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+            first_entity_token_spans = [(entity_token_start, entity_token_end + 2)]
+
+        elif self.task == "entity_pair_classification":
+            assert (
+                isinstance(entity_spans, list)
+                and len(entity_spans) == 2
+                and isinstance(entity_spans[0], tuple)
+                and isinstance(entity_spans[1], tuple)
+            ), "Entity spans should be provided as a list of tuples, each tuple containing the start and end character indices of an entity"
```
Same here.

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+        elif self.task == "entity_span_classification":
+            mask_entity_id = self.entity_vocab["[MASK]"]
+
+            assert isinstance(entity_spans, list) and isinstance(
+                entity_spans[0], tuple
+            ), "Entity spans should be provided as a list of tuples, each tuple containing the start and end character indices of an entity"
```
Same here.

sgugger commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+            return BatchEncoding(encoded_inputs, tensor_type=return_tensors)
+
+        batch_size = len(required_input)
+        assert all(
+            len(v) == batch_size for v in encoded_inputs.values()
+        ), "Some items in the output dictionary have a different batch size than others."
```
Same here.

sgugger commented on tests/test_tokenization_mluke.py:
```diff
@@ -0,0 +1,591 @@
+        encoded_sentence = tokenizer.build_inputs_with_special_tokens(text)
+        encoded_pair = tokenizer.build_inputs_with_special_tokens(text, text_2)
+
+        assert encoded_sentence == encoded_text_from_decode
+        assert encoded_pair == encoded_pair_from_decode
```
Use `self.assertEqual` here.

sgugger reviewed (approved):
Nice addition! I just left a few comments on the style, mainly all the `assert`.

qqaatw committed:
Improve tokenizer tests (#13594)

* Use new method to acquire tokenizers

* Resolve TODOs.

* Style

* Fix

* Enable do_lower_case in test_tokenize_special_tokens

* Apply suggestion from code review

* Fix mask token handling

* Revert "Fix mask token handling"

This reverts commit daaa3f5291b1f71e5bc3604ca281c000000c4648.

* Fix FNet mask token tokenization

* Complete everything

* Apply suggestions from code review

stas00 committed:
[CI] move env print to util, add pt, nccl versions (#14607)

* move env print to util, add pt, nccl versions

* style

* version

* align

LysandreJik commented on src/transformers/models/luke/modeling_luke.py:
```diff
@@ -674,6 +717,38 @@ def forward(self, hidden_states):
         return pooled_output
 
 
+class EntityPredictionHeadTransform(nn.Module):
```
This could benefit from a `# Copied from` statement originating from others, like BERT!

LysandreJik commented on src/transformers/models/mluke/tokenization_mluke.py:
```diff
@@ -0,0 +1,1581 @@
+        self._eventual_warn_about_too_long_sequence(encoded_inputs["input_ids"], max_length, verbose)
+
+        # Padding
+        # To do: add padding of entities
```
Should this be done before merging?

Ryou0634 replied:
Padding entity inputs seems already implemented, so maybe this is just unintentionally left in the code that I copied it from.
I’ve added some tests to make sure padding is correctly done just in case.

LysandreJik reviewed (approved):
This looks good to me! Thank you, @Ryou0634

LysandreJik committed:
2022 is the year of multi-modality (#14610)

* 2022 is the year of multi-modality

* Small fix

* Apply suggestions from code review

Co-authored-by: Suraj Patil <surajp815@gmail.com>
Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>
Co-authored-by: Anton Lozhkov <aglozhkov@gmail.com>

* Apply suggestions from code review

* Apply to documentation index

* Apply suggestions from code review

Co-authored-by: lewtun <lewis.c.tunstall@gmail.com>

* Update README.md

Co-authored-by: lewtun <lewis.c.tunstall@gmail.com>

* Apply suggestions from code review

* Apply suggestions from code review

Co-authored-by: Suraj Patil <surajp815@gmail.com>
Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>
Co-authored-by: Anton Lozhkov <aglozhkov@gmail.com>
Co-authored-by: lewtun <lewis.c.tunstall@gmail.com>

LysandreJik committed:
Fix doc builder (#14616)

* Fix doc builder

* Fix doc builder

* Fix doc builder

stas00 committed:
[trainer] add tf32-mode control (#14606)

* [trainer] add --tf32 support

* it's pt>=.17

* it's pt>=.17

* flip the default to True

* add experimental note

* simplify logic

* style

* switch to 3-state logic

* doc

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* re-style code

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Rocketknight1 committed:
Make DefaultDataCollator importable from root (#14588)

* Make DefaultDataCollator importable from root

* Add documentation for DefaultDataCollator and add return_tensors argument to all class docstrings

* make style

* Add DefaultDataCollator to data_collator.rst

* Add DefaultDataCollator to data_collator.rst

yuchenlin committed:
fix a typo (#14626)

ryokan0123 committed:
Apply suggestions from code review

Fix typo

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

ryokan0123 committed:
Apply suggestions from code review

Fix typo

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

ryokan0123 committed:
Update src/transformers/models/luke/modeling_luke.py

fix style

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

ryokan0123 committed:
Update src/transformers/models/luke/modeling_luke.py

fix style

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

ryokan0123 committed:
update docs

kamalkraj committed:
updated readme with proper arguments (#14624)

sgugger commented:
I think there was something wrong in your rebase, and the diff now shows much more than what you are actually changing. Could you close your PR and reopen a fresh one from your branch?

patil-suraj committed:
Add Flax example tests (#14599)

* add test for glue

* add tests for clm

* fix clm test

* add summrization tests

* more tests

* fix few tests

* add test for t5 mlm

* fix t5 mlm test

* fix tests for multi device

* cleanup

* ci job

* fix metric file name

* make t5 more robust

patil-suraj committed:
fix typo (#14635)

ryokan0123 committed:
replace assert with ValueError for Luke and mLUKETokenizer and add tests

ryokan0123 committed:
replace assert with ValueError for Luke and mLUKETokenizer and add tests

ryokan0123 committed:
Merge branch 'add_luke_multilingual' of github.com:Ryou0634/transformers into add_luke_multilingual

ryokan0123 committed:
add Copied from... to MLukeTokenizer

ryokan0123 committed:
(M)LukeTokenizer is now able to handle padding with sentneces with no entity

ryokan0123 committed:
Delete unnecessary comments

ryokan0123 committed:
add Copied from in modeling_luke.py

Ryou0634 commented:
Sorry, seems like I messed it up.
I will reopen a new pull request.
